ADR/####-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,280 @@
+  - Operational actions: triggering builds/tests, starting manual releases, pipeline management
+  - Any other runtime actions that don't involve declarative resource configuration
+- **Commit to single onboarding flow**: UI-based onboarding will be completely deprecated with no fallback option to minimize maintenance overhead and deliver optimal user outcomes
+- **Maintain ease of use and excellent user experience** by providing a VS Code plugin that delivers forms based configuration of validation of tenant resources
+
+### IDE Selection Rationale
+VS Code was chosen as the primary IDE target because:
+- **Wide adoption**: VS Code is extensively used across development teams
+- **Robust plugin ecosystem**: Provides comprehensive APIs for linting, schema validation, and custom tooling
+- **Advanced visualization capabilities**: Plugin system supports rich visualizations and can run full, sandboxed React-based applications within webview windows
+- **Extensibility**: Enables building sophisticated onboarding wizards and configuration interfaces directly within the familiar IDE environment
+
+### Component Architecture
+
+| Component | Role | Optional |
+|-----------|------|----------|
+| VS Code Plugin | Main interface for onboarding and configuration | Yes |
+| Git Repos | Source of truth for Konflux config | Yes |
+| GitOps Registration Service | API for automated registration of GitOps repos as tenants | **Yes** (recommended) |
+| ArgoCD Service | Continuous GitOps sync and deployment | **Yes** (recommended) |
+| Konflux UI | Monitoring, runtime insights, and operational actions (non-configuration) | Yes |
+
+### VS Code Plugin Design
+
+**Core Features:**
+- Forms and wizards to scaffold and edit YAML objects
+- Built-in schema-aware YAML editor
+- Linting and validation against Konflux schemas
+- Git-aware diff view and PR-ready commit generation
+- Optional: preview pipeline graph / dependency resolution
+
+
+**GitOps Flow:**
+
+1. Developer creates an empty git repo with Github, Gitlab, etc.
+4. **If first time**: Uses wizard to register repo with Tenant Registration Service (declares namespace name, configures ArgoCD)
+5. Developer is prompted by UI to install / open repo in VS Code
+3. Launches Konflux plugin and uses it to configure new components, tests, releasePlans
+6. Plugin generates YAML objects in repo
+7. Developer commits and pushes code (optionally via PR, with CI,reviews etc.)
+9. ArgoCD automatically syncs merged changes to the namespace
+10. sync status is reported back to PR, if there was one
+
+### Multi-Repo GitOps Model & Optional Services
+
+**GitOps Registration Service (Optional):**
+Konflux will optionally provide a **GitOps Registration Service** API that automates the registration and lifecycle of GitOps repositories as first-class tenants. This service is recommended for production deployments but not required for basic Konflux functionality.
+
+**ArgoCD Service (Optional):**
```

Comment:
As of now, we already have some infinite loops between cluster and ArgoCD. They are caused by invalid yamls in user's GitOps repositories. It creates a lot of unwanted workload.
I think we should agree and document, that ArgoCD (or another GitOps tool) should not endlessly re-sync, but do it on commit or user request.

Reply:
Maybe also give user ability to trigger sync manually from the UI/cli? So there is a simple and easy way how to trigger sync if the automatic one failed.

Reply:
isn't there some way to tell argo not to keep retrying to sync something which already failed? this seems preventable